### PR TITLE
[frontend] init grammar

### PR DESCRIPTION
**Pull request by Bryanskiy**

### CODE REVIEW

File: include/frontend/frontend.hh [hunk 1]
```diff
@@ -19,14 +21,21 @@ public:
...
 private:
-  std::unique_ptr<yyFlexLexer> m_lexer = nullptr;
+  void reportSyntaxError(const parser::context &ctx);
+  void reportExpctdTok(const yy::parser::context &ctx);
+  void reportUnexpctdTok(const yy::parser::context &ctx);
+
+  static inline constexpr size_t numTokens = 10;
+
+private:
+  std::unique_ptr<Lexer> m_lexer = nullptr;
```
Comment (Tako-San) [inline include/frontend/frontend.hh]: We have another naming convention for private members: lexer_

File: include/frontend/lLexer.hh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#include <cctype>
+#include <iostream>
+
+#ifndef yyFlexLexer
+#include <FlexLexer.h>
+#endif /* yyFlexLexer */
+
+#include "parser.hh"
+
+class Lexer : public yyFlexLexer {
```
Comment (Tako-San) [inline include/frontend/lLexer.hh]: Add final

File: include/frontend/lLexer.hh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  Lexer(const Lexer &flx) = delete;
+  Lexer &operator=(const Lexer &) = delete;
+
+  Lexer(Lexer &&flx) = delete;
+  Lexer &operator=(Lexer &&) = delete;
+
+  Lexer(std::istream &in, std::ostream &out) : yyFlexLexer{in, out} {}
+
+  yy::location getCurLocation();
```
Comment (Tako-San) [inline include/frontend/lLexer.hh]: Should it be const?

File: src/frontend/lLexer.cc [hunk 4]
```diff
@@ -0,0 +1,35 @@
+#include "frontend/lLexer.hh"
+
+yy::location Lexer::getCurLocation()
+{
```
Comment (Tako-San) [inline src/frontend/lLexer.cc]: Not formatted

File: src/frontend/lLexer.cc [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+int Lexer::getLastLine() const
+{
+  return lastNumOfLine;
+}
+
+void Lexer::updCurLoc()
+{
+  auto cur_num_of_line = lineno();
```
Comment (Tako-San) [inline src/frontend/lLexer.cc]: We use camel style

**Review by Tako-San — CHANGES_REQUESTED** [review]

**Review by derzhavin3016 — APPROVED** [review]
LGBTQIA+

**Review by Tako-San — APPROVED** [review]
LGTM

**Pull request merged**